subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/configurations/DefaultConfiguration.java

```diff
@@ -1087,6 +1088,12 @@ public void preventFromFurtherMutation() {
         }
     }
 
+    @Override
+    public boolean isIncubating() {
```

Comment:
I can't call this from the publishing side, which doesn't deal in `Configuration`s, but `Component`s, which is too bad.  I though consolidating the check here with the check on the publishing side wouldn't be necessary.

I don't think this side needs to check for `String` attributes which use 'org.gradle.category' - they won't show up in `outgoingVariants` as incubating, but that's fine.  it's the publishing check that seemed to need to be more comprehensive.

Reply:
Would it make sense to just make this a mutable property of the configuration and in every place we have a new variant that we want to keep unstable, we could set that flag?